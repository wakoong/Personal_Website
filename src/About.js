import React from 'react';
import profile from './assets/images/Woosik.png'

class About extends React.Component {
    render() {
        return(
            <div className="main-background">
                <div className="main-block">
                    <div className="profile-image">
                        <img src={profile} alt="instagram" />
                    </div>
                    <div className="description">
                    Welcome!
                    </div>
                </div>
            </div>
        )
    }
}

export default About;