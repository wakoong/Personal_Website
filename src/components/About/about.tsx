import React from 'react';

class About extends React.Component {
  render() {
    return (
      <div className="about">
        <div className="about-image" />
        <section>
          <p>👋 Welcome!</p>
          <p>
            👨🏻‍🔬 This is my "lab" where I get to experiment various things and
            showcase my mini projects. Things can go pretty crazy at times but I
            manage to eventually fix them.
          </p>
          <p>
            👨🏻‍💻 I am an enthusiatic learner who likes documenting progress. Here
            is the link to my official blog website{' '}
            <a href="https://www.donkeykoong.com" target="_blank">
              www.donkeykoong.com
            </a>
            . You can visit my website to find out why it's called donkeykoong.
            🤔
          </p>
          <p>
            🏀 I love basketball #ballislife{' '}
            <a href="https://www.instagram.com/woosikka/" target="_blank">
              instagram icon goes here
            </a>
          </p>
        </section>
      </div>
    );
  }
}

export default About;
