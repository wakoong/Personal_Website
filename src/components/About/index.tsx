import React from 'react';

import instagram from '../../assets/images/instagram.png';

class About extends React.Component {
  render() {
    return (
      <div className='about'>
        <section>
          <div className='image' />
          <p className='title'>👋 Welcome!</p>
          <p>
            👨🏻‍🔬 This is my "lab" where I get to experiment various things and showcase my mini projects. Things can go
            pretty crazy at times but I manage to eventually fix them.
          </p>
          <p>
            👨🏻‍💻 I am an enthusiatic learner who likes documenting progress. My main programming language is JavaScript
            and I specialize in ReactJS for a frontend framework. I also have some experience with Python and Django. I
            am recently interested in learning more about Node.js, MongoDB, GraphQL, and algorithms.
          </p>
          <p>🏀 Also, I love basketball and learning new human languages.</p>
          <p className='bold'>"Optimisim is a moral choice. Cynicism is the last refuge for cowards" - Jim Yong Kim</p>
          <br />
          <br />
        </section>
      </div>
    );
  }
}

export default About;
