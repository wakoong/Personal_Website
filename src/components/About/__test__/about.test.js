import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import About from '../about.tsx';

afterEach(cleanup);

test('renders with snapshot', () => {
  const { asFragment } = render(<About />);
  expect(asFragment()).toMatchSnapshot();
});

test('Contains welcome text with hand wavy emoji', () => {
  const { getByText } = render(<About />);
  expect(getByText('ðŸ‘‹ Welcome!')).toBeInTheDocument();
});
