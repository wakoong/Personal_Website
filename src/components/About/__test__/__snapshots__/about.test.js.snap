// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with snapshot 1`] = `
<DocumentFragment>
  <div
    class="about"
  >
    <div
      class="about-image"
    />
    <section>
      <p>
        👋 Welcome!
      </p>
      <p>
        👨🏻‍🔬 This is my "lab" where I get to experiment various things and showcase my mini projects. Things can go pretty crazy at times but I manage to eventually fix them.
      </p>
      <p>
        👨🏻‍💻 I am an enthusiatic learner who likes documenting progress. Here is the link to my official blog website 
        <a
          href="https://www.donkeykoong.com"
          target="_blank"
        >
          www.donkeykoong.com
        </a>
        . You can visit my website to find out why it's called donkeykoong. 🤔
      </p>
      <p>
        🏀 I love basketball #ballislife 
        <a
          href="https://www.instagram.com/woosikka/"
          target="_blank"
        >
          instagram icon goes here
        </a>
      </p>
    </section>
  </div>
</DocumentFragment>
`;
